### PR TITLE
Add multi-language support for speech input on web.

### PR DESCRIPTION
**Pull request by superhindupur**
## Purpose
Required by #549. Without these changes, the chat frontend only recognizes english.  This PR adds support for multi-lingual speech recognition from the Chat frontend. Currently the additional languages are hardcoded into the frontend code, however, the plan under #317 is to move this list of languages to the deployment configuration in the near future. 

This PR also introduces the first frontend unit test with the required scaffolding and support to extend this test suite. Hereafter the expectation for frontend changes will be that they are accompanied by one or more unit tests. 

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Deploy CWYD, and speak through the microphoine input in one of the following languages: French, German, Italian. Your words should be converted correctly to text. 

To run the frontend unit tests, checkout the code and run 
```
make unittest-frontend
```

### CODE REVIEW

File: code/frontend/.mocharc.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (superhindupur) [inline code/frontend/.mocharc.json:1]: https://typestrong.org/ts-node/docs/recipes/mocha/ is the official recipe for typescript + mocha. 

We need ts-node for mocha to work with typescript. We are setting the loader to ESM so that it works with the ECMAScript modules in our codebase.

File: code/frontend/src/pages/chat/Chat.tsx [hunk 2]
```diff
@@ -168,12 +167,12 @@ const Chat = () => {
...
-          serviceRegion
-        );
-        const audioConfig = AudioConfig.fromDefaultMicrophoneInput();
-        const recognizer = new SpeechRecognizer(speechConfig, audioConfig);
+        const recognizer = multiLingualSpeechRecognizer(subscriptionKey, serviceRegion, [
+          "en-US",
+          "fr-FR",
+          "de-DE",
+          "it-IT"
+        ]);
```
Comment (superhindupur) [inline code/frontend/src/pages/chat/Chat.tsx:175]: This file is doing too much, and has many many imports. I needed to test the multi-lingual configuration for the speech recognizer, so I extracted it into another file, and tested that file. As we're adding new functionality to CWYD, we should try and modularize this code so that each file is of a manageable size and complexity and is conducive for testing.

Comment (superhindupur) [inline code/frontend/src/pages/chat/Chat.tsx:175]: Also, as mentioned on the PR description, the plan is to move this hardocded list of languages into the deployment config under #317 .

File: Makefile [hunk 3]
```diff
@@ -38,9 +38,9 @@ uitest: ## 🧪 Run the ui tests in headless mode
 	@echo -e "\e[34m$@\e[0m" || true
 	@cd tests/integration/ui && npm install && npx cypress run --env ADMIN_WEBSITE_NAME=$(ADMIN_WEBSITE_NAME),FRONTEND_WEBSITE_NAME=$(FRONTEND_WEBSITE_NAME)
 
-build-frontend: ## 🏗️ Build the Frontend webapp
+unittest-frontend: ## 🏗️ Unit test the Frontend webapp
```
Comment (adamdougal) [inline Makefile:41]: Did you mean to replace `build-frontend` with this? It might be worth having both `build` and `unittest` tasks for the front end.

File: code/frontend/src/util/SpeechToText.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    subscriptionKey: string,
+    serviceRegion: string,
+    languages: string[]
+  ) => {
+        const speechConfig = SpeechConfig.fromSubscription(
+          subscriptionKey,
+          serviceRegion
+        );
+        const audioConfig = AudioConfig.fromDefaultMicrophoneInput();
+        var autoDetectSourceLanguageConfig = AutoDetectSourceLanguageConfig.fromLanguages(languages)
```
Comment (adamdougal) [inline code/frontend/src/util/SpeechToText.tsx:19]: Nit: Is there any reason this is a `var` and the rest are `const`? Probably doesn't matter either way, just noticed it was different!

**Review by adamdougal — CHANGES_REQUESTED** [review]

File: Makefile (hunk 3 above)
Comment (superhindupur) [reply Makefile:41]: I replaced it because it followed the same pattern as we have for the python code - we don't have an explicit "build" command, we only have a unit-test. 

I thought the only reason we had the build-frontend was because there weren't any tests to run. Do you see any benefit in keeping the build-frontend?

File: code/frontend/src/util/SpeechToText.tsx (hunk 4 above)
Comment (superhindupur) [reply code/frontend/src/util/SpeechToText.tsx:19]: Oops, missed it when changing the others to const. Will edit this, thanks!

File: code/frontend/.npmignore
Comment (cecheta) [inline code/frontend/.npmignore:1]: Is this file required? I thought it was for when publishing an npm package

**Comment by cecheta** [comment]
Could we get a ticket to include these tests in the GitHub Action?

File: Makefile (hunk 3 above)
Comment (superhindupur) [reply Makefile:41]: lol, my brain is super-wired to java. Ok, I don't feel very strongly on the subject so I don't mind reinstating the build-frontend command, but I will leave the ci command to call the unittest-frontend command directly.

File: code/frontend/.npmignore
Comment (superhindupur) [reply code/frontend/.npmignore:1]: You're right, I was assuming that this is what keeps the test files from being included in the final index.js asset created by npm run build - but no, it turns out to be the include: ["src"] setting in the tsconfig.json file (https://www.typescriptlang.org/tsconfig#include).

I removed the .npmigore file now. I double checked that the test file isn't included in npm run build by writing invalid code in it and npm run build still succeeded (whereas npm run test failed)

**Comment by superhindupur** [comment]
> Could we get a ticket to include these tests in the GitHub Action?

@cecheta I added this as a subtask of #317

**Review by adamdougal — APPROVED** [review]
lgtm

**Review by cecheta — APPROVED** [review]
Nice 😄
